File: c680(computer_scoring_1)/Python/computer_scoring.py
from sys import stdin, stdout


for line in stdin:
    num_of_problems = int(line)
    correct_answer: str = stdin.readline()
    num_of_students = int(stdin.readline())

    # stores the answers in str and compare them directly by indexing
    for _ in range(num_of_students):
        answer: str = stdin.readline()
        correct: int = 0
        # too much not count, not enough no penalty
        for i in range(num_of_problems if len(answer) > num_of_problems else len(answer)):
            if answer[i] == correct_answer[i]:
                correct += 1
        stdout.write(f'{(correct * 100) // num_of_problems}\n')

File: c680(computer_scoring_1)/Java/ComputerScoring.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;


public class ComputerScoring {
    public static void main(String[] args) throws IOException {
        /* try-with-resources */
        try (BufferedReader in = new BufferedReader(new InputStreamReader(System.in))) {
            int numberOfProblems = Integer.parseInt(in.readLine());
            /* stores the answers in str and compare them directly by indexing */
            char[] correctAnswer = in.readLine().toCharArray();
            int numOfStudents = Integer.parseInt(in.readLine());

            while ((numOfStudents--) != 0) {
                char[] answer = in.readLine().toCharArray();
                /* counter */
                int correct = 0;
                /* too much not count, not enough no penalty */
                int valids = (numberOfProblems > correctAnswer.length
                              ? correctAnswer.length
                              : numberOfProblems);
                for (int i = 0; i < valids; ++i) {
                    if (answer[i] == correctAnswer[i]) {
                        ++correct;
                    }
                }
                System.out.println(correct * 100 / numberOfProblems);
            }
        }
    }
}
